### ISSUE TITLE
[Bug]: PHP Setting Page not Loaded 

### ISSUE THREAD

**Issue by fftfaisal** [beyondcode/herd-community#973]
### Platform

Windows

### Operating system version

Windows 11 Pro 23H2 (22631.4037)

### System architecture

Windows

### Herd Version

1.10.0

### PHP Version

php 8.3

### Bug description

I can not change the PHP setting, because it's infinitely loading the page. I tried to re-install the herd but still same. 

![image](https://github.com/user-attachments/assets/3e933e53-c447-4693-ae3c-96cfd4ab81aa)

### Relevant log output

```shell
2024-09-10 13:07:18.894 +06:00 [INF] Starting to read the hosts file.
2024-09-10 13:07:18.896 +06:00 [INF] Ensuring hosts file permissions.
2024-09-10 13:07:18.897 +06:00 [INF] Logging hosts file permissions.
2024-09-10 13:07:18.898 +06:00 [INF] Permissions for C:\Windows\System32\drivers\etc\hosts:
2024-09-10 13:07:18.898 +06:00 [INF] Identity: NT AUTHORITY\SYSTEM, Access Control Type: Allow, Rights: FullControl, Inheritance: None, Propagation: None
2024-09-10 13:07:18.898 +06:00 [INF] Identity: BUILTIN\Administrators, Access Control Type: Allow, Rights: FullControl, Inheritance: None, Propagation: None
2024-09-10 13:07:18.898 +06:00 [INF] Identity: BUILTIN\Users, Access Control Type: Allow, Rights: ReadAndExecute, Synchronize, Inheritance: None, Propagation: None
2024-09-10 13:07:18.898 +06:00 [INF] Identity: APPLICATION PACKAGE AUTHORITY\ALL APPLICATION PACKAGES, Access Control Type: Allow, Rights: ReadAndExecute, Synchronize, Inheritance: None, Propagation: None
2024-09-10 13:07:18.898 +06:00 [INF] Identity: APPLICATION PACKAGE AUTHORITY\ALL RESTRICTED APPLICATION PACKAGES, Access Control Type: Allow, Rights: ReadAndExecute, Synchronize, Inheritance: None, Propagation: None
2024-09-10 13:07:18.898 +06:00 [INF] Current user: NT AUTHORITY\SYSTEM
2024-09-10 13:07:18.898 +06:00 [INF] Current user access rights:
2024-09-10 13:07:18.898 +06:00 [INF] Access rights for current user (NT AUTHORITY\SYSTEM): FullControl
2024-09-10 13:07:18.898 +06:00 [ERR] Reading content of hosts file
2024-09-10 13:07:18.898 +06:00 [INF] 127.0.0.1 database.herd.test
2024-09-10 13:07:18.898 +06:00 [INF] 127.0.0.1 ielts.test
2024-09-10 13:07:18.898 +06:00 [INF] 127.0.0.1 adminlte.test
2024-09-10 13:07:18.898 +06:00 [INF] 127.0.0.1 affiliate.test
2024-09-10 13:07:18.898 +06:00 [INF] 127.0.0.1 chatify.test
2024-09-10 13:07:18.898 +06:00 [INF] 127.0.0.1 eddy.test
2024-09-10 13:07:18.898 +06:00 [INF] 127.0.0.1 elang.test
2024-09-10 13:07:18.898 +06:00 [INF] 127.0.0.1 eprap.test
2024-09-10 13:07:18.898 +06:00 [INF] 127.0.0.1 filament.test
2024-09-10 13:07:18.898 +06:00 [INF] 127.0.0.1 genealogy.test
2024-09-10 13:07:18.898 +06:00 [INF] 127.0.0.1 ielts.test
2024-09-10 13:07:18.898 +06:00 [INF] 127.0.0.1 jetstrap-vuexy.test
2024-09-10 13:07:18.898 +06:00 [INF] 127.0.0.1 lawyer.test
2024-09-10 13:07:18.898 +06:00 [INF] 127.0.0.1 lawyers.test
2024-09-10 13:07:18.899 +06:00 [INF] 127.0.0.1 lms.test
2024-09-10 13:07:18.899 +06:00 [INF] 127.0.0.1 mailer.test
2024-09-10 13:07:18.899 +06:00 [INF] 127.0.0.1 metronic9.test
2024-09-10 13:07:18.899 +06:00 [INF] 127.0.0.1 new.test
2024-09-10 13:07:18.899 +06:00 [INF] 127.0.0.1 qbank.test
2024-09-10 13:07:18.899 +06:00 [INF] 127.0.0.1 reverb.test
2024-09-10 13:07:18.899 +06:00 [INF] 127.0.0.1 reverb-chat.test
2024-09-10 13:07:18.899 +06:00 [INF] 127.0.0.1 samsonqbank.test
2024-09-10 13:07:18.899 +06:00 [INF] 127.0.0.1 samsonqbanks.test
2024-09-10 13:07:18.899 +06:00 [INF] 127.0.0.1 server.test
2024-09-10 13:07:18.899 +06:00 [INF] 127.0.0.1 stripe-payment.test
2024-09-10 13:07:18.899 +06:00 [INF] 127.0.0.1 team.test
2024-09-10 13:07:18.899 +06:00 [INF] 127.0.0.1 video.test
2024-09-10 13:07:18.899 +06:00 [INF] 127.0.0.1 vito.test
2024-09-10 13:07:18.899 +06:00 [INF] 127.0.0.1 vuexy.test
2024-09-10 13:07:18.899 +06:00 [INF] 127.0.0.1 wordpress.test
2024-09-10 13:07:18.899 +06:00 [INF] Herd-specific sections found. Beginning update.
2024-09-10 13:07:18.899 +06:00 [INF] Writing updated contents to the hosts file.
2024-09-10 13:07:18.900 +06:00 [INF] Restored original ACL for hosts file.
2024-09-10 13:07:18.900 +06:00 [INF] The hosts file was successfully updated.
2024-09-10 13:07:18.900 +06:00 [INF] Hosts file updated with received content.
2024-09-10 17:49:30.884 +06:00 [INF] Request to /status endpoint.
2024-09-10 17:49:33.190 +06:00 [INF] Starting to read the hosts file.
2024-09-10 17:49:33.190 +06:00 [INF] Ensuring hosts file permissions.
2024-09-10 17:49:33.190 +06:00 [INF] Logging hosts file permissions.
2024-09-10 17:49:33.191 +06:00 [INF] Permissions for C:\Windows\System32\drivers\etc\hosts:
2024-09-10 17:49:33.191 +06:00 [INF] Identity: NT AUTHORITY\SYSTEM, Access Control Type: Allow, Rights: FullControl, Inheritance: None, Propagation: None
2024-09-10 17:49:33.191 +06:00 [INF] Identity: BUILTIN\Administrators, Access Control Type: Allow, Rights: FullControl, Inheritance: None, Propagation: None
2024-09-10 17:49:33.191 +06:00 [INF] Identity: BUILTIN\Users, Access Control Type: Allow, Rights: ReadAndExecute, Synchronize, Inheritance: None, Propagation: None
2024-09-10 17:49:33.191 +06:00 [INF] Identity: APPLICATION PACKAGE AUTHORITY\ALL APPLICATION PACKAGES, Access Control Type: Allow, Rights: ReadAndExecute, Synchronize, Inheritance: None, Propagation: None
2024-09-10 17:49:33.191 +06:00 [INF] Identity: APPLICATION PACKAGE AUTHORITY\ALL RESTRICTED APPLICATION PACKAGES, Access Control Type: Allow, Rights: ReadAndExecute, Synchronize, Inheritance: None, Propagation: None
2024-09-10 17:49:33.191 +06:00 [INF] Current user: NT AUTHORITY\SYSTEM
2024-09-10 17:49:33.191 +06:00 [INF] Current user access rights:
2024-09-10 17:49:33.191 +06:00 [INF] Access rights for current user (NT AUTHORITY\SYSTEM): FullControl
2024-09-10 17:49:33.191 +06:00 [ERR] Reading content of hosts file
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 database.herd.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 adminlte.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 affiliate.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 chatify.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 eddy.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 elang.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 eprap.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 filament.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 genealogy.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 ielts.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 jetstrap-vuexy.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 lawyer.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 lawyers.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 lms.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 mailer.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 metronic9.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 new.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 qbank.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 reverb.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 reverb-chat.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 samsonqbank.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 samsonqbanks.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 server.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 stripe-payment.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 team.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 video.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 vito.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 vuexy.test
2024-09-10 17:49:33.192 +06:00 [INF] 127.0.0.1 wordpress.test
2024-09-10 17:49:33.192 +06:00 [INF] Herd-specific sections found. Beginning update.
2024-09-10 17:49:33.192 +06:00 [INF] Writing updated contents to the hosts file.
2024-09-10 17:49:33.192 +06:00 [INF] Restored original ACL for hosts file.
2024-09-10 17:49:33.192 +06:00 [INF] The hosts file was successfully updated.
2024-09-10 17:49:33.192 +06:00 [INF] Hosts file updated with received content.
2024-09-10 17:49:35.911 +06:00 [INF] Request to /status endpoint.
2024-09-10 17:49:42.774 +06:00 [INF] Request to /status endpoint.
2024-09-10 17:54:35.906 +06:00 [INF] Application is shutting down...
2024-09-10 17:54:35.907 +06:00 [DBG] Hosting stopping
2024-09-10 17:54:35.914 +06:00 [INF] HTTP Server stopped.
2024-09-10 17:54:35.917 +06:00 [DBG] Hosting stopped
2024-09-10 17:55:50.761 +06:00 [DBG] Hosting starting
2024-09-10 17:55:50.812 +06:00 [INF] Config path: C:\ProgramData\Herd\service.json.
2024-09-10 17:55:50.814 +06:00 [INF] Config file with port exists, using port: 5000.
2024-09-10 17:55:50.814 +06:00 [INF] Application started. Hosting environment: Production; Content root path: C:\Users\USER\.config\herd\bin\
2024-09-10 17:55:50.816 +06:00 [DBG] Hosting started
2024-09-10 17:55:50.826 +06:00 [INF] HTTP Server started on http://localhost:5000/.
2024-09-10 18:03:48.404 +06:00 [INF] Request to /status endpoint.
2024-09-10 18:03:53.429 +06:00 [INF] Request to /status endpoint.
2024-09-10 18:03:58.430 +06:00 [INF] Request to /status endpoint.
2024-09-10 18:04:03.427 +06:00 [INF] Request to /status endpoint.
2024-09-10 18:04:08.429 +06:00 [INF] Request to /status endpoint.
```

**Comment by sschlein** [comment]
I guess Herd has no access to the internet because it's blocked by a firewall and so it can't reach the API for PHP versions. We have that fixed in the upcoming update and even without an internet connection, this site loads.

**Comment by mpociot** [comment]
This is now fixed in Herd 1.11.0
